src/app/util/ember-compatibility-functions.cpp

```diff
@@ -728,6 +745,19 @@ CHIP_ERROR WriteSingleClusterData(ClusterInfo & aClusterInfo, TLV::TLVReader & a
         return apWriteHandler->AddStatus(attributePathParams, Protocols::InteractionModel::Status::NeedsTimedInteraction);
     }
 
+    {
+        Access::SubjectDescriptor subjectDescriptor; // TODO: get actual subject descriptor
```

Comment:
Per spec this check happens before the timed write check (but after the IsReadOnly check).  Please move this code accordingly and add a TODO to add a test for the ordering.

Reply:
I put the check in the correct spot. It's easier here for the write, than for the read, because the attribute override is in a better spot. We should clean up the attribute override in read attribute so it's similarly easier to do this. And yes I'm anticipating we'll be testing the proper stages of errors as we get more of this functionality in place.